File: authority-portal-backend/authority-portal-quarkus/src/main/kotlin/de/sovity/authorityportal/broker/services/api/filtering/CatalogSearchService.java
/*
 *  Copyright (c) 2023 sovity GmbH
 *
 *  This program and the accompanying materials are made available under the
 *  terms of the Apache License, Version 2.0 which is available at
 *  https://www.apache.org/licenses/LICENSE-2.0
 *
 *  SPDX-License-Identifier: Apache-2.0
 *
 *  Contributors:
 *       sovity GmbH - initial API and implementation
 *
 */

package de.sovity.authorityportal.broker.services.api.filtering;

import de.sovity.authorityportal.broker.dao.pages.catalog.CatalogQueryFields;
import de.sovity.authorityportal.broker.dao.utils.SearchUtils;
import jakarta.enterprise.context.ApplicationScoped;
import lombok.RequiredArgsConstructor;
import org.jooq.Condition;

import java.util.List;

@ApplicationScoped
public class CatalogSearchService {

    public Condition filterBySearch(CatalogQueryFields fields, String searchQuery) {
        return SearchUtils.simpleSearch(searchQuery, List.of(
            fields.getDataOfferTable().ASSET_ID,
            fields.getDataOfferTable().ASSET_TITLE,
            fields.getDataOfferTable().DATA_CATEGORY,
            fields.getDataOfferTable().DATA_SUBCATEGORY,
            fields.getDataOfferTable().DESCRIPTION,
            fields.getDataOfferTable().CURATOR_ORGANIZATION_NAME,
            fields.getDataOfferTable().KEYWORDS_COMMA_JOINED,
            fields.getConnectorTable().ENDPOINT_URL,
            fields.getConnectorTable().MDS_ID,
            fields.getOrganizationName()
        ));
    }
}

File: authority-portal-backend/authority-portal-quarkus/src/main/kotlin/de/sovity/authorityportal/broker/services/api/filtering/CatalogSearchService.kt
/*
 *  Copyright (c) 2023 sovity GmbH
 *
 *  This program and the accompanying materials are made available under the
 *  terms of the Apache License, Version 2.0 which is available at
 *  https://www.apache.org/licenses/LICENSE-2.0
 *
 *  SPDX-License-Identifier: Apache-2.0
 *
 *  Contributors:
 *       sovity GmbH - initial API and implementation
 *
 */
package de.sovity.authorityportal.broker.services.api.filtering

import de.sovity.authorityportal.broker.dao.pages.catalog.CatalogQueryFields
import de.sovity.authorityportal.broker.dao.utils.SearchUtils.simpleSearch
import jakarta.enterprise.context.ApplicationScoped
import org.jooq.Condition

@ApplicationScoped
class CatalogSearchService {
    fun filterBySearch(fields: CatalogQueryFields, searchQuery: String?): Condition {
        return simpleSearch(
            searchQuery, listOf(
                fields.dataOfferTable.ASSET_ID,
                fields.dataOfferTable.ASSET_TITLE,
                fields.dataOfferTable.DATA_CATEGORY,
                fields.dataOfferTable.DATA_SUBCATEGORY,
                fields.dataOfferTable.DESCRIPTION_NO_MARKDOWN,
                fields.dataOfferTable.KEYWORDS_COMMA_JOINED,
                fields.connectorTable.ENDPOINT_URL,
                fields.connectorTable.MDS_ID,
                fields.organizationTable.NAME
            )
        )
    }
}
